include/mbedtls/ssl_internal.h

```diff
@@ -238,7 +238,7 @@
    implicit sequence number. */
 #define MBEDTLS_SSL_HEADER_LEN 13
 
-#if defined(MBEDTLS_SSL_DTLS_CONNECTION_ID)
+#if !defined(MBEDTLS_SSL_DTLS_CONNECTION_ID)
```

Comment:
In commit "Fix I/O buffer sizes when MBEDTLS_SSL_DTLS_CONNECTION_ID is enabled", we say "Update related buffer size calculations.", but there is no such code change in the patch. Also, consider making the commit early in your patch set (could be the first commit in this set, as it just fixes a pre-existing bug).